docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -598,7 +598,7 @@ instantiated during the augment process and utilize <<injection,injection>>.
 Build items are concrete, final subclasses of the abstract `io.quarkus.builder.item.BuildItem` class.  Each build item represents
 some unit of information that must be passed from one stage to another.  The base `BuildItem` class may not itself be directly
 subclassed; rather, there are abstract subclasses for each of the kinds of build item subclasses that _may_ be created:
-<<simple-build-items,simple>>, <<multi-build-items,multi>>, and <<empty-build-items,empty>>.
+<<simple-build-items,simple>>, <<multi-build-items,multi>>, <<empty-build-items,empty>>, <<validation-error-build-item,validation>> and <<artifact-result-build-item,artifact>>
```

Comment:
> there are abstract subclasses [...] validation and artifact

Those are not abstract class, and regardless, this paragraph is specifically about explaining the difference between simple/multi/empty, so I don't think that's the right place to add links to `ValidationErrorBuildItem`/`ArtifactResultBuildItem`.

I think this change should be removed.

Maybe you could instead add a note (starting with `NOTE: `) to the section about build steps, explaining that build steps only get executed if they produce build items that are (transitively) needed to build the application, and that a build step that doesn't produce anything should probably produce `ValidationErrorBuildItem`/`ArtifactResultBuildItem` instead?